src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGenerationException.java

```diff
@@ -36,14 +36,4 @@ public MetadataGenerationException(String message) {
         super(message);
     }
 
-    /**
-     * Creates a new {@code MetadataGenerationException} instance
-     * that was caused by another exception.
-     *
-     * @param message a message that explains this exception
-     * @param cause the Exception that is wrapped by this exception
-     */
-    public MetadataGenerationException(String message, Throwable cause) {
```

Comment:
for next PR:

I just questioning whole reason of this Exception class existence.
if we want to give user marker on what kind of exception happen (to let them easily catch it) .... why we do not use this exception to wrap all exceptions during metadata load ?
so this ctor become git a logical place.

if we do not care to give users special exception ... let use IllegalState exception instead this this custom runtime expection

Reply:
It is a common practice to create custom exceptions for specific cases. This metadata generation tool is a sandalone stuff from main checkstyle functionality, so case seems appropriate to me. Wrapping of exceptions sounds good, I will try to do this next time.

IllegalState is not for this kind of failures.
from javadoc
>  Signals that a method has been invoked at an illegal or
 inappropriate time.  In other words, the Java environment or
 Java application is not in an appropriate state for the requested
 operation.

So custom exception class name says enough to understand that issue is related to metadata generation rather than abstract "illegal state"

Reply:
>IllegalState is not for this kind of failures.

We do this very frequently. To avoid messing with custom exceptions. Users unlikely to read documentation. Usually the most reliable way is to wrap third-party with catch all exception block  to not let any exception damage application.